### ISSUE TITLE
Dismissing toasts sometimes causes crash.

### ISSUE THREAD

**Issue by alexmercerind** [alexmercerind/desktoasts#2]
Not sure of the cause right now.

**Comment by alexmercerind** [comment]
Found the cause, will fix later.

**Comment by alexmercerind** [comment]
Turns out, this actually is because of `dispose` calls in the `Toast` class. Do not call it unless you're sure that user has dismissed all the notifications. The library will work fine if you take care of this.

I decided to delete the object with `dispose` method since I take memory management in C++ seriously. Although, calling `dispose` from `ToastService` is safe once you're sure that you're not going to send any new notifications during lifecycle of the application.

This exception is caused in two ways:
- Either you `dispose` that `Toast` object before user dismissed it. Thus, the `Toast::toastTemplate` is actually a `nullptr` & crash is caused.
- The notification was sent by previous running instance of your application & user dismisses it. Thus when user dismisses an old toast when a new instance is running, the `Toast::toastTemplate` is actually a garbage value & crash is caused.

This is a design bug in `WinToast` & possibly only occurs when event handler is set up as in this library's case there is.
I'll make PR to `WinToast` and this will be ~~fixed~~ avoided.

Thanks.
